numba/tests/test_unicode.py

```diff
@@ -357,6 +361,33 @@ def test_slice3_error(self):
                                          cfunc(s, sl),
                                          "'%s'[%d:%d:%d]?" % (s, i, j, k))
 
+    def test_zfill(self):
+        pyfunc = zfill_usecase
+        cfunc = njit(pyfunc)
+
+        ZFILL_INPUTS = [
+            'ascii',
+            '+ascii',
+            '-ascii',
+            '-asc ii-',
+            '12345',
+            '-12345',
+            '+12345',
+            '',
+            '¡Y tú crs?',
+            '🐍⚡',
+            '大眼，小手。',
```

Comment:
It'd be good to have a kind-2 or kind-4 unicode type with a `+/-` as the leading char.

Reply:
@stuartarchibald Do you think it is enough to just add `+` and `-` to these strings? I don't know how to write `+` in kind-4 unicode

Reply:
I'd suggest copying the 3 strings above and for each doing:
```
original_string,
+original_string,
-original_string,
```
that should be sufficient, thanks.

Reply:
@stuartarchibald done it in latest commit "adding more tests"

Reply:
great, thanks